clientv3/watch.go

```diff
@@ -371,6 +371,9 @@ func (w *watcher) Close() (err error) {
 			err = werr
 		}
 	}
+	if err == context.Canceled {
+		return fmt.Errorf("Context canceled as expected")
```

Comment:
I feel like we should return nil here as context cancel is expected with a related message as here, so that end user see call to the Close() as success instead of one returned with error. Just a thought as we are discussing the PR. Thanks!